### PR TITLE
Import Script and Buildpath fix

### PR DESCRIPTION
**Pull request by Nitsorn**
can import scripts in production apps, and append build to filepath only when unpacked

### CODE REVIEW

File: app/components/PlatformAppWebview.vue.ts [hunk 1]
```diff
@@ -67,13 +69,18 @@ export default class PlatformAppWebview extends Vue {
   }
 
   get appUrl() {
+    console.log('appUrl: ', this.platformAppsService.getPageUrlForSlot(
+      this.appId,
+      this.pageSlot
+    ));
```
Comment (andycreeth) [inline app/components/PlatformAppWebview.vue.ts]: can we remove

File: app/components/PlatformAppWebview.vue [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <template>
 <webview
-  v-if="renderWebview"
+  v-if="renderWebview && appUrl"
```
Comment (andycreeth) [inline app/components/PlatformAppWebview.vue]: Is there any valid scenario where we wouldn't have a url here?

File: app/services/platform-apps/index.ts [hunk 3]
```diff
@@ -16,6 +16,7 @@ import { HostsService } from 'services/hosts';
 import { handleErrors, authorizedHeaders } from 'util/requests';
 import { UserService } from 'services/user';
 import { trim, compact } from 'lodash';
+import { cd } from 'shelljs';
```
Comment (andycreeth) [inline app/services/platform-apps/index.ts]: This looks like an automatic import lol

File: app/components/PlatformAppWebview.vue.ts [hunk 4]
```diff
@@ -67,13 +69,18 @@ export default class PlatformAppWebview extends Vue {
...
     return this.platformAppsService.getPageUrlForSlot(
       this.appId,
       this.pageSlot
     );
   }
 
   get appPartition() {
+    console.log('App partition: ', this.platformAppsService.getAppPartition(this.appId));
```
Comment (andycreeth) [inline app/components/PlatformAppWebview.vue.ts]: Can be removed?

File: app/services/platform-apps/index.ts [hunk 5]
```diff
@@ -134,6 +135,7 @@ export class PlatformAppsService extends
   appUnload = new Subject<string>();
 
   private localStorageKey = 'PlatformAppsUnpacked';
+  private baseCdnUrl = 'https://s3.amazonaws.com/streamlabs-platform';
```
Comment (andycreeth) [inline app/services/platform-apps/index.ts]: We don't want to hard code this.  We want to add `appUrl` which you added last week as part of your production app stuff: https://github.com/stream-labs/streamlabs-obs/pull/840/files#diff-5ff6bf525e7ffbac00adf915bb8e26f2R99

Essentially here's the logic:
- Is this app running unpacked?  Do nothing past what we're normally doing
- Is this app running in production?  Make sure any script being loaded has the same beginning of the URL as `appUrl` we got back from the server.  That ensures the script belongs to this app.

File: app/services/platform-apps/index.ts [hunk 6]
```diff
@@ -183,6 +185,7 @@ export class PlatformAppsService extends
  */
   async installProductionApps() {
     const productionApps = await this.fetchProductionApps();
+    console.log('production apps', productionApps);
```
Comment (andycreeth) [inline app/services/platform-apps/index.ts]: remove

File: app/services/platform-apps/index.ts [hunk 7]
```diff
@@ -468,20 +471,25 @@ export class PlatformAppsService extends
         }
 
         if (details.resourceType === 'script') {
-          console.log('got details', details);
-          console.log('GOT REQUEST', details.url);
+          // for cdn scripts, resourceType is other
```
[the remark this answers is not in the document]
Comment (Nitsorn) [reply app/services/platform-apps/index.ts]: oh no, will remove thanks

File: app/services/platform-apps/index.ts [hunk 8]
```diff
@@ -254,7 +254,7 @@ export class PlatformAppsService extends
     this.appLoad.next(this.getApp(id));
   }
 
-  async validateManifest(manifest: IAppManifest, appPath: string) {
+  async validateManifest(manifest: IAppManifest, appPath: string, isUnpacked = true) {
```
[the remark this answers is not in the document]
Comment (Nitsorn) [reply app/services/platform-apps/index.ts]: gotcha, fixing

**Pull request merged**